apps/worker/tasks/base.py

```diff
@@ -248,37 +312,62 @@ def safe_retry(self, max_retries=None, countdown=None, exc=None, **kwargs):
             **kwargs: Additional kwargs to pass to self.retry()
 
         Returns:
-            True if retry was scheduled
-            False if max retries exceeded
+            False if max retries exceeded (returns immediately)
+            Raises Retry exception if retry was scheduled (this is intentional -
+            the Retry exception propagates to Celery to schedule the retry)
+
+        Note:
+            The `return True` statement is intentionally unreachable because
+            `self.retry()` raises a `Retry` exception that propagates to Celery.
+            This is the correct behavior - callers should use try-except if they
+            need to handle the retry case differently.
 
         Example:
             if some_condition_requires_retry:
                 if not self.safe_retry(max_retries=5, countdown=60):
                     # Max retries exceeded
                     log.error("Giving up after too many retries")
                     return {"success": False, "reason": "max_retries"}
+                # If retry was scheduled, Retry exception propagates and this code won't execute
         """
         current_retries = self.request.retries if hasattr(self, "request") else 0
+        total_attempts = self._get_total_attempts()
         task_max_retries = max_retries if max_retries is not None else self.max_retries
 
-        if task_max_retries is not None and current_retries >= task_max_retries:
-            log.error(
-                f"Task {self.name} exceeded max retries",
-                extra={
-                    "task_name": self.name,
-                    "current_retries": current_retries,
-                    "max_retries": task_max_retries,
-                },
-            )
-            TASK_MAX_RETRIES_EXCEEDED_COUNTER.labels(task=self.name).inc()
-            return False
+        # Check both retry count and total attempts to catch visibility timeout re-deliveries
+        # Total attempts = retries + 1 (initial attempt), so we compare against max_retries + 1
+        max_total_attempts = (
+            task_max_retries + 1 if task_max_retries is not None else None
```

Comment:
there is a lot of +1 stuff going on in this PR, is there a way to eliminate it by using better models for the vars?